woocommerce/woocommerce-ios
# Google Ads: Navigation from Hub menu entry point

Closes: #13218 


## Description

This PR handles the navigation for GLA from the Hub menu entry point. Changes include:
- Updated `DefaultGoogleAdsEligibilityChecker` to optimize the check following the technical post pe5sF9-2O6-p2.
- Updated `HubMenuViewModel` to check if the site has campaigns to determine the web page to display:
  - If the site has existing campaigns, display the dashboard page.
  - Otherwise, display the campaign creation page.
- Updated `WebView` and `AuthenticatedWebView` to support `exitTrigger` without `urlToTriggerExit`.
- Added `GoogleAdsCampaignCoordinator` to handle navigation to Google ads campaigns.
- Updated `HubMenu` and `HubMenuHostingController` to start the coordinator when tapping the Google ads entry point.

## Testing steps

- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Log in to your test store and navigate to the Menu tab.
- Select Google for WooCommerce. Confirm that a web view is presented for campaign management.
  - If the store is a self-hosted store authenticated with WPCom credentials, you'd be asked to log in with WPOrg credentials. 
  - If the store is a WPCom store or self-hosted store authenticated with WPOrg credentials, you'd be authenticated automatically.
- Assuming your site has no paid campaign created yet, tapping the entry point should display the creation flow. Proceed to create a test campaign.
- When the campaign is created successfully, the web view should be dismissed.
- Tap on the entry point again, this time you should see the campaign dashboard page.


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/209da385-6a52-4c53-a56d-1dbb4bba8927



---
